Specification.md

```diff
@@ -1,17 +1,145 @@
-## Gas limit
+# Specification
 
-By default `web3.js` uses `web3.eth.Contract.method.myMethod().estimateGas()`, but this value can differ from the 
-actual gas that will be used, if the contract you are calling depends on the blockhash, blocknumber or any other source of 
-randomness making the gas cost nondeterministic.
+The CLI should package several "core" extensions.
+The API these extensions expose is aimed at `node` environments.
+The aim is to provide convenience to power-users & devs.
 
-E.g.: the [`MiniMeToken`](https://github.com/aragon/aragon-apps/blob/master/shared/minime/contracts/MiniMeToken.sol) contract which snapshots balances at certain block numbers.
+## IPFS `@aragon/ipfs-utils`
 
-Considering the above-mentioned behavior, the CLI should calculate the recommended gas limit as follows:
-- `recommendedGas = estimatedGas`, if `estimatedGas > upperGasLimit` or
+Goals:
+
+- installing & starting a daemon with the right configuration
+- interacting with local/remote nodes: authentication, configuration, pinning, etc.
+- help setup a "production" node (?)
+- automatically pin anything published to an APM repository (?)
+
+### IPFS Binaries
+
+Local node:
+
+- ✔️ `aragon ipfs` - Alias for `aragon ipfs start`
```

Comment:
I've added a bunch of potential changes here after brainstorming a bit about the `ipfs` and `devchain` commands. Would love some feedback! cc: @0xGabi @sohkai @izqui 

I think people usually "optimize" their process and run `ipfs` and `devchain` in a separate terminal, so that `aragon run` and other commands are faster. This is why I think we should not stop IPFS/Devchain as part of these commands, but instead expose commands to do so.

Example 
> IPFS not started, do you wanna start it? [yes/no] 
Remember to do `aragon ipfs stop` afterwards.

And that `ipfs start` should finish and leave the daemon running in the background.

Reply:
I agree, having management commands for the daemon would be great! It'd also solve things like the race condition in https://github.com/aragon/aragon-cli/issues/383.